File: app/src/main/java/com/kaciula/archiman/presentation/screen/home/HomeComponent.java
package com.kaciula.archiman.presentation.screen.home;

import com.kaciula.archiman.util.injection.ActivityScope;
import com.kaciula.archiman.util.injection.AppComponent;
import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = HomeModule.class)
public interface HomeComponent {

  void inject(HomeController homeController);

  void inject(UserDialogFragment userDialogFragment);
}

File: app/src/main/kotlin/com/kaciula/archiman/presentation/screen/home/HomeComponent.kt
package com.kaciula.archiman.presentation.screen.home

import com.kaciula.archiman.injection.ActivityScope
import com.kaciula.archiman.injection.AppComponent
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(HomeModule::class))
interface HomeComponent {

    fun inject(homeController: HomeController)

    fun inject(userDialogFragment: UserDialogFragment)
}
